test/contactagent/AgendaTest.java

```diff
@@ -0,0 +1,116 @@
+package contactagent;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class AgendaTest {
+
+    @Test
+    public void showHowClientCanMessUp() {
+        Agenda agenda = new Agenda();
+        agenda.addContact(new Contact("Dany", "2691677"));
+        assertTrue(agenda.existsContact("Dany"));
+
+        Contact[] list = agenda.listContacts();
+        list[0].setName("Nathalia");
+
+        assertFalse(agenda.existsContact("Dany"));
+    }
+
+    @Test
+    public void addOneContact() {
+        Agenda agenda = new Agenda();
+
+        agenda.addContact(new Contact("Dany", "2691677"));
+
+        assertTrue(agenda.existsContact("Dany"));
+
+    }
+
+    @Test
+    public void addExistingContact() {
+        Agenda agenda = new Agenda();
+
+        agenda.addContact(new Contact("Kristina", "12345"));
+        agenda.addContact(new Contact("Kristina", "12345"));
+        assertTrue(agenda.existsContact("Kristina"));
+    }
+
+    @Test
+    public void addTwoContacts() {
+        Agenda agenda = new Agenda();
+
+        agenda.addContact(new Contact("Dany", "2691677"));
+        agenda.addContact(new Contact("Caro", "4848760"));
+
+        assertTrue(agenda.existsContact("Dany"));
+        assertTrue(agenda.existsContact("Caro"));
+    }
+
+    @Test
+    public void contactNotAdded() {
+        Agenda agenda = new Agenda();
+
+        agenda.addContact(new Contact("Natalia", "12344"));
+
+        assertFalse(agenda.existsContact("Dany"));
+
+    }
+
+    @Test
+    public void listTheContacts() {
+        Agenda agenda = new Agenda();
+        //I want this array of 3 contact [3]
```

Comment:
remove this comment. You should review your own pull request and delete the "garbage"